tests/functionality/pflow/selfcheck.cpp

```diff
@@ -34,52 +34,76 @@ struct PflowFunctionalityTests
   using Params = PflowFunctionalityTestParameters;
   MPI_Comm comm;
   int nprocs;
+  int logging_rank = 0;
 
   PflowFunctionalityTests(std::string testsuite_filename, MPI_Comm comm,
                           int logging_verbosity = EXAGO_LOG_INFO)
       : FunctionalityTestContext(testsuite_filename, logging_verbosity),
         comm{comm} {
+    int my_rank;
+    auto rerr = MPI_Comm_rank(comm, &my_rank);
+    if (rerr)
+      throw ExaGOError("Error getting MPI rank number");
+
     auto err = MPI_Comm_size(comm, &nprocs);
-    if (err)
-      throw ExaGOError("Error getting MPI num ranks");
+    if (err) {
+      if (my_rank == logging_rank)
+        throw ExaGOError("Error getting MPI num ranks");
+      exit(0);
+    }
```

Comment:
Again I don't think `logging_rank` is relevant here as it's just a 0

Reply:
I don't think I was the one that originally put in logging_rank (I could be wrong, my memory is going). @jaelynlitz did you add this? It looks like some kind of future-proofing to me.

Reply:
IMO just a 0 everywhere is fine, but we could make a global variable if we really want

Reply:
Yeah I think I originally put in the `logging_rank`. I mean I would argue for more configurable code instead of hard coding to 0 but it works either way

Reply:
but yes should be consistent one way or the other

Reply:
I think 0 is a sane default, and if you want we can have this in a private internal ExaGO header.

Reply:
I just got rid of logging_rank and replaced it with zero wherever it showed up. I will be pretty easy to add it back in in the future if there is a need for it. Since IO demands in this code are pretty low, just assuming it is 0 is probably a good bet.